holoviews/operation/downsample.py

```diff
@@ -188,6 +188,7 @@ def _m4(x, y, n_out, **kwargs):
             'The m4 downsampling algorithm requires the tsdownsampler '
             'library to be installed.'
         ) from None
+    n_out = n_out - (n_out % 4)  # n_out must be a multiple of 4
```

Comment:
Wouldn't it be safer to _increase_ the number of samples, to handle degenerate cases where otherwise there might be only a tiny number of samples? Please ignore if I'm misunderstanding this.

Reply:
We never downsample something where the number of points is smaller than the width of the plot, so there will never be a tiny number of samples. 

The case reported in the issue `n_out` is around 210 to 217 samples, so adding or subtracting 3 at most for it seemed negligible.